syntax = "proto3";

package unaryposts;

service UnaryPosts{
 rpc create_post(Create) returns (MessageResponse) {}
 rpc update_post(Update) returns (MessageResponse) {}
 rpc delete_post(Delete) returns (MessageResponse) {}
 rpc get_post(Get) returns (PostResponse) {}
 rpc get_posts(GetAll) returns (PostsResponse) {}
}

message Create{
 string user = 1;
 string text = 2;
}

message MessageResponse{
 string message = 1;
}

message Update{
 string user = 1;
 int32 id = 2;
 string text = 3;
}

message Delete{
 string user = 1;
 int32 id = 2;
}

message Get{
 int32 id = 1;
}

message GetAll{
 string user = 1;
}

message PostResponse{
 string message = 1;
 string user = 2;
}

message Posts{
 int32 id = 1;
 string text = 2;
}

message PostsResponse{
 repeated Posts posts = 1;
}
